cypress/integration/edit_segment.spec.js

```diff
@@ -0,0 +1,384 @@
+/// <reference types="cypress" />
+// Tests the ability to edit a segment
+
+describe("Edit segment", () => {
+    beforeEach(() => {
+    
+        cy.visit(Cypress.env('login_url'));
+        cy.get("[name=csrfmiddlewaretoken]")
+          .should("exist")
+          .should("have.attr", "value")
+          .as("csrfToken");
+    
+        cy.get("@csrfToken").then((token) => {
+          cy.request({
+            method: "POST",
+            url: Cypress.env("login_url"), 
+            form: true,
+            body: {
+              username: "cypress",
+              password: "1234asdf",
+            },
+            headers: {
+              "X-CSRFTOKEN": token,
+            },
+          });
+        });
```

Comment:
Protip! If you have do this in a lot of tests, especially across multiple different files, I'd consider moving this to a [Cypress custom command](https://docs.cypress.io/api/cypress-api/custom-commands.html). e.g,. so that the `beforeEach()` would just be a single line:

```js
 cy.loginAs("cypress")
```

Reply:
Oh that's beautiful. So far, this is the only place where being logged in actually changes your permissions.